src/state/graphicalItemsSlice.ts

```diff
@@ -42,6 +43,14 @@ export type CartesianGraphicalItemSettings = {
    * One graphical item can have multiple error bars. This probably only makes sense in Scatter.
    */
   errorBars: ReadonlyArray<ErrorBarsSettings> | undefined;
+  stackId: StackId | undefined;
+  /**
+   * Why not just stop pushing the graphical items to state when they are hidden?
+   * Well some components decide to continue showing them anyway.
```

Comment:
right - this is useful for having the `hide` prop control visibility on legend click. `includeHidden` as well - if the graphical item wasn't there it couldn't use that dataset to keep the domain the same